src/Medidata.ZipkinTracer.Core/TraceProvider.cs

```diff
@@ -120,8 +120,19 @@ public ITraceProvider GetNext()
         /// <returns>true: parsed</returns>
         private bool Parse(string value)
         {
-            long result;
-            return !string.IsNullOrWhiteSpace(value) && Int64.TryParse(value, NumberStyles.HexNumber, CultureInfo.InvariantCulture, out result);
+            ulong result;
+            return !string.IsNullOrWhiteSpace(value) && UInt64.TryParse(value, NumberStyles.HexNumber, CultureInfo.InvariantCulture, out result);
+        }
+
+        /// <summary>
+        /// Get Lower 16 Characters of an id
+        /// </summary>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        private string GetLower16Characters(string value)
+        {
+            if (string.IsNullOrWhiteSpace(value)) return value;
+            return value.Length > 16 ? value.Substring(value.Length - 16) : value;
         }
```

Comment:
Question, is this trying to return the first or the last 16 characters?
E.g., if value = "abcdefghijklmnopq", what should be returned?